x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/AbstractBuilder.java

```diff
@@ -137,7 +137,8 @@ public static String unquoteString(Source source) {
         checkForSingleQuotedString(source, text, 0);
 
         text = text.substring(1, text.length() - 1);
-        StringBuffer resultString = new StringBuffer();
+        text = handleUnicodeChars(source, text);
+        StringBuilder resultString = new StringBuilder();
         Matcher regexMatcher = slashPattern.matcher(text);
```

Comment:
decoupling these two might be incorrect.
how does `foo \\u{0041}` get unescaped for example? should just be the`\\` that gets unescaped to `\`. and `foo \\\u{0041}` gets unescaped to `foo \A`

Reply:
Thx, I have to revisit this.

Reply:
Pushed the change: https://github.com/elastic/elasticsearch/pull/70514/commits/0dcb1c325dc410aa9d03fab481ea0ee4da936857
and thx again for pointing it out!

Reply:
saw the new tests, thanks!